### PR TITLE
feat(1712): Relaxing blockedBy for same job

### PR DESCRIPTION
**Pull request by klu909**
## Context

adding the option for job blocking behavior
## Objective

adding the option for job blocking behavior
## References

https://github.com/screwdriver-cd/screwdriver/issues/1712
## License

I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: config/job.js [hunk 1]
```diff
@@ -163,6 +163,12 @@ const SCHEMA_BLOCKEDBY_VALUE = Joi.alternatives().try(
 );
 const SCHEMA_BLOCKEDBY = Joi.alternatives().try(Joi.array().items(SCHEMA_BLOCKEDBY_VALUE), SCHEMA_BLOCKEDBY_VALUE);
 const SCHEMA_FREEZEWINDOWS = Joi.alternatives().try(Joi.array().items(SCHEMA_CRON_EXPRESSION), SCHEMA_CRON_EXPRESSION);
+const SCHEMA_BLOCKEDBYSAMEJOB = Joi.boolean().optional();
+const SCHEMA_BLOCKEDBYSAMEJOBWAITTIME = Joi.number()
+    .integer()
+    .positive()
+    .optional()
+    .allow(null);
```
Comment (tkyi) [inline config/job.js:171]: should we add defaults here?

Comment (klu909) [reply config/job.js:171]: default is handled in queue-service, but yeah we can add it here too

File: config/job.js [hunk 2]
```diff
@@ -180,6 +186,8 @@ const SCHEMA_JOB = Joi.object()
         description: SCHEMA_DESCRIPTION,
         environment: SCHEMA_ENVIRONMENT,
         freezeWindows: SCHEMA_FREEZEWINDOWS,
+        blockedBySameJob: SCHEMA_BLOCKEDBYSAMEJOB,
+        blockedBySameJobWaitTime: SCHEMA_BLOCKEDBYSAMEJOBWAITTIME,
```
Comment (tkyi) [inline config/job.js:190]: it kind of seems weird to me that it's a job-level field vs annotation, but I guess `blockedBy` is also at this level

I think I would prefer these fields as annotations

Comment (klu909) [reply config/job.js:190]: Reasoning is that you probably wouldn't want this on every job

Comment (tkyi) [reply config/job.js:190]: could be job-level annotation, right?

**Review by tkyi — APPROVED** [review]

**Pull request merged**